test/utils/testconfig/testconfig.go

```diff
@@ -0,0 +1,274 @@
+package testconfig
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"context"
+
+	ptpv1 "github.com/openshift/ptp-operator/api/v1"
+	testclient "github.com/openshift/ptp-operator/test/utils/client"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ConfigCheck                  = "CONFIGCHECK"
+	OrdinaryClockString          = "OC"
+	BoundaryClockString          = "BC"
+	DualNICBoundaryClockString   = "DualNICBC"
+	DiscoveryString              = "Discovery"
+	NoneString                   = ""
+	StartString                  = ""
+	InitStatusString             = "init"
+	ConfiguredStatusString       = "configured"
+	DiscoverySuccessStatusString = "discoverySuccess"
+	DiscoveryFailureStatusString = "discoveryFailure"
+)
+
+type ConfigStatus int64
+
+const (
+	Start ConfigStatus = iota
+	InitStatus
+	ConfiguredStatus
+	DiscoverySuccessStatus
+	DiscoveryFailureStatus
+)
+
+type PTPMode int64
+
+const (
+	OrdinaryClock PTPMode = iota
+	BoundaryClock
+	DualNICBoundaryClock
+	Discovery
```

Comment:
Will this change exiting discovery env available?  it  is used by cn-test

Reply:
yes, the original env DISCOVERY_MODE=true/false is replaced by PTP_TEST_MODE=OC/BC/DualNICBC/Discovery

Reply:
AS you are aware of cnf-test is one of the consumers of these tests. We need to retain the original env variable since this will break  cnf-test when we propagate these  changes to that project 
https://github.com/openshift-kni/cnf-features-deploy/blob/7a47e1d408013bc375abd8f8420c9a6488b88736/cnf-tests/testsuites/e2esuite/README.md

https://github.com/openshift-kni/cnf-features-deploy/blob/c5e4f1a70e472230c796921bfe86e068bd10c17a/cnf-tests/README.md#running-tests-in-discovery-mode


https://github.com/openshift-kni/cnf-features-deploy/blob/c5e4f1a70e472230c796921bfe86e068bd10c17a/cnf-tests/README.md#ptp-tests

Reply:
fixed